### PR TITLE
feat[CL]: add num initialized ticks query

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

We currently have a LiquidityNetInDirection query, which allows us to provide a start and end tick, traverse this range, and return the liquidity net. The unknown here is how many initialized ticks actually exist in this range. 

For purposes of understanding the impact a swap will have on the current sqrt price, it's helpful to request the next X initialized ticks and return their liquidity depths, regaurdless of how many ticks need to be traversed to get to X.

## Testing and Verifying

TBD

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

**Comment by czarcas7ic** [comment]
devbot help

**Comment by devbot-wizard** [comment]
Hi! I'm DevBot, a bot that helps with common tasks in the development process. Commands:
- `devbot fix`: Fix basic errors in the PR. (e.g. imports, changelog conflicts, go mod conflicts)
- `devbot merge base`: Merge the base branch into the PR branch.
- `devbot add changelog [misc/feat/bug/api/sec] [message]`: Add a changelog entry to the PR. (e.g. `devbot add changelog feat Added a new feature`)
    - If message is blank, defaults to PR title.
- `devbot re-pr`: Re-PR the PR, useful for external contributors where we have no edit perms.

**Comment by czarcas7ic** [comment]
devbot add changelog feat

File: x/concentrated-liquidity/query.go [hunk 1]
```diff
@@ -244,3 +244,95 @@ func (k Keeper) getTickByTickIndex(ctx sdk.Context, poolId uint64, tickIndex int
...
+	var iterator db.Iterator
+	if zeroForOne {
+		iterator = prefixStore.ReverseIterator(boundTickKey, startTickKey)
+	} else {
+		iterator = prefixStore.Iterator(startTickKey, storetypes.InclusiveEndBytes(boundTickKey))
+	}
+
+	liquidityDepths := []queryproto.TickLiquidityNet{}
+
+	iterationCount := uint64(0)
```
Comment (p0mvn) [inline x/concentrated-liquidity/query.go:315]: Is this still needed?

**Review by p0mvn — APPROVED** [review]
LGTM

File: x/concentrated-liquidity/query.go (hunk 1 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/query.go:315]: Yes, iterationCount refers to the user provided literation limit

Comment (p0mvn) [reply x/concentrated-liquidity/query.go:315]: Ah right, mistook this for the Go-side limit we discussed before

**Pull request merged**